bindings/cpp/session.cpp

```diff
@@ -2287,6 +2340,110 @@ async_iterator_result session::cancel_iterator(const key &id, uint64_t iterator_
 	return iterator(id, data);
 }
 
+async_iterator_result session::server_send(const std::vector<std::string> &keys, uint64_t iflags, const std::vector<int> &groups)
+{
+	if (get_groups().empty()) {
+		async_iterator_result result(*this);
+		async_result_handler<iterator_result_entry> handler(result);
+		handler.complete(create_error(-ENXIO, "server_send: local group list is empty"));
+		return result;
+	}
+
+	if (groups.empty()) {
+		async_iterator_result result(*this);
+		async_result_handler<iterator_result_entry> handler(result);
+		handler.complete(create_error(-ENXIO, "server_send: remote group list is empty"));
+		return result;
+	}
+
+	if (keys.empty()) {
+		async_iterator_result result(*this);
+		async_result_handler<iterator_result_entry> handler(result);
+		handler.complete(create_error(-ENXIO, "server_send: key list is empty"));
+		return result;
+	}
+
+	int local_group = get_groups().front();
```

Comment:
I know that we have the same in all iterator methods, but we have to add to iterator methods' and server_send's descriptions that they use only first group from groups specified in the session.